src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -120,17 +120,15 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 		let globallyIgnored = <string[]>JSON.parse(this.storageService.get('extensionsAssistant/ignored_recommendations', StorageScope.GLOBAL, '[]'));
 		this._globallyIgnoredRecommendations = globallyIgnored.map(id => id.toLowerCase());
 
-		this.loadRecommendationsPromise = this.getWorkspaceRecommendations()
-			.then(() => {
-				// these must be called after workspace configs have been refreshed.
-				this.fetchCachedDynamicWorkspaceRecommendations();
-				this.fetchFileBasedRecommendations();
-				this.fetchExperimentalRecommendations();
-				return this.promptWorkspaceRecommendations();
-			}).then(() => {
-				this._modelService.onModelAdded(this.promptFiletypeBasedRecommendations, this, this._disposables);
-				this._modelService.getModels().forEach(model => this.promptFiletypeBasedRecommendations(model));
-			});
+		this.fetchCachedDynamicWorkspaceRecommendations();
+		this.fetchFileBasedRecommendations();
+		this.fetchExperimentalRecommendations();
```

Comment:
While moving the above out of the wait for workspace recommendations promise does seem to fix the issue, it isnt clear why it does so.

`queryLocal` adds recommendation sources only after `getAllReccomendations` promise is resolved. So why is the bug occuring in the first place?

Reply:
As discussed, `getAllRecommendations` only returns the value of the `this._fileBasedRecommendations` variable which will be empty unless we make this change.

So this fix is good